docs/assets/diagrams/linking/2-8-sign-challenge.puml

```diff
@@ -0,0 +1,139 @@
+@startuml
+
+title: Sign Challenge
+
+hide footbox
+
+box "Mobile Device" #Light
+  actor Alice
+  participant "PISP App" as App
+end box
+box "PISP" #Light
+  participant "Firebase" as Firebase
+  participant "PISP Server" as Server
+end box
+box "Mojaloop" #Light
+  participant "Switch" as Mojaloop
+end box
+
+'start flow
+Alice -> App ++: Give fingerprint
+
+rnote right of App #Light
+  ""firebase.firestore()""
+  ""  .collection("consents")""
+  ""  .where("consentId", "==", "123")""
+  ""  .set({""
+  ""    credential: {""
+  ""      id: "12345",""
+  ""      challenge: {""
+  ""        signature: base64(...)""
```

Comment:
The payload here may be something like (from app -> firebase and firebase -> pisp-demo-server). Although I cannot quite confirm yet that this will be the finalized format.

Registration:
```
clientDataJSON: ( ... )
attestationObject: ( ... )
```
Signing:
```
clientDataJson: (...)
authenticatorData: (...)
signature: (...)
```

Reply:
The demo server may have to then encode/decode/format these in a way that is palatable by the `auth-service`

Reply:
can i check whether the signing part is for the transfer flow?

Reply:
Yep, signing is part of the transfer flow. The consent registration requires those objects I sent above though.

Reply:
let's keep this open to be updated in the future iteration once we have a clearer decision on this :)